src/Config/TerminusConfig.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Pantheon\Terminus\Config;
+
+use Pantheon\Terminus\Exceptions\TerminusException;
+
+/**
+ * Class TerminusConfig
+ * @package Pantheon\Terminus\Config
+ */
+class TerminusConfig extends \Robo\Config
+{
+    /**
+     * @var string
+     */
+    protected $constant_prefix = 'TERMINUS_';
+
+    /**
+     * @var array
+     */
+    protected $sources = [];
+
+    /**
+     * Set add all the values in the array to this Config object.
+     * @param array $array
+     */
+    public function fromArray($array) {
+        foreach ($array as $key => $val) {
+            $this->set($key, $val);
+        }
+    }
+
+    /**
+     * Convert the config to an array.
+     * @return array
+     */
+    public function toArray() {
+        $out = [];
+        foreach ($this->keys() as $key) {
+            $out[$key] = $this->get($key);
+        }
+        return $out;
+    }
+
+    /**
+     * Set a config value. Converts key from Terminus constant (TERMINUS_XXX) if needed.
+     *
+     * @param string $key
+     * @param mixed $value
+     *
+     * @return $this
+     */
+    public function set($key, $value) {
+        // Convert Terminus constant name to internal key.
+        if ($this->keyIsConstant($key)) {
+            $key = $this->getKeyFromConstant($key);
+        }
+        return parent::set($key, $value);
+    }
+
+    /**
+     * Fet a configuration value
```

Comment:
Ah, clearly this should be replaced with the full `field-effect transformer`... :trollface: